src/Core/Checkout/Customer/SalesChannel/LoginRoute.php

```diff
@@ -135,10 +134,6 @@ public function login(RequestDataBag $data, SalesChannelContext $context): Conte
             $this->rateLimiter->reset(RateLimiter::LOGIN_ROUTE, $cacheKey);
         }
 
-        if (!$customer->getActive()) {
-            throw new InactiveCustomerException($customer->getId());
```

Comment:
This exception should still be thrown in cases where the customer registered with double-opt-in required and when the customer has not yet confirmed the opt-in. Please leave the if clause and adapt the filter closure to still include inactive customers if they still require opt-in:
```php
if ($customer->getDoubleOptInRegistration() && !$customer->getDoubleOptInConfirmDate()) {
    return true;
}
```
Also add tests for this case please.